File: python/src/world.py
import numpy as np

from .settings import SIZE

cube = [
    np.array([0, 0, 0]),
    np.array([1, 0, 0]),
    np.array([0, 1, 0]),
    np.array([0, 0, 1]),
    np.array([1, 1, 0]),
    np.array([1, 0, 1]),
    np.array([0, 1, 1]),
    np.array([1, 1, 1]),
]

class World:
    def __init__(self):
        self.walls = self.generate()

    def generate(self):
        walls = list()
        for wall in range(6):
            for x in range(SIZE):
                for y in range(SIZE):
                    if wall == 0:
                        # Top
                        walls.append(np.array([x, SIZE + 1, y]))
                    if wall == 1:
                        # Bot
                        walls.append(np.array([x, -1, y]))
                    if wall == 2:
                        # Left
                        walls.append(np.array([-1, x, y]))
                    if wall == 3:
                        # Right
                        walls.append(np.array([SIZE + 1, x, y]))
                    if wall == 4:
                        # Front
                        walls.append(np.array([x, y, SIZE + 1]))
                    if wall == 5:
                        # Back
                        walls.append(np.array([x, y, -1]))

        return walls

    def render(self):
        # TODO make time-based oscillation, pass deltatime as parameter
        return make_cubes(self.walls)

def make_cubes(positions):
    return [make_cube(position) for position in positions]

def make_cube(position):
    return [vertex + position for vertex in [vertex - np.array([0.5, 0.5, 0.5]) for vertex in cube]]

File: project/src/world.cpp
#include <vector>

#include <glm/glm.hpp>

#include "settings.hpp"

#include "world.hpp"

World::World() {
  // Top / Bottom
  // Overreach both sides
  for (int x = -1; x < int(Settings::Size) + 1; x++) {
    for (int z = -1; z < int(Settings::Size) + 1; z++) {
      wall.push_back(glm::vec3(x, int(Settings::Size) + 1, z));
    }
  }
  for (int x = -1; x < int(Settings::Size) + 1; x++) {
    for (int z = -1; z < int(Settings::Size) + 1; z++) {
      wall.push_back(glm::vec3(x, -1, z));
    }
  }

  // Right / Left
  // Overreach Z axis
  for (int y = 0; y < int(Settings::Size); y++) {
    for (int z = -1; z < int(Settings::Size) + 1; z++) {
      wall.push_back(glm::vec3(int(Settings::Size) + 1, y, z));
    }
  }
  for (int y = 0; y < int(Settings::Size); y++) {
    for (int z = -1; z < int(Settings::Size) + 1; z++) {
      wall.push_back(glm::vec3(-1, y, z));
    }
  }

  // Front / Back
  // No overreaching
  for (int x = 0; x < int(Settings::Size); x++) {
    for (int y = 0; y < int(Settings::Size); y++) {
      wall.push_back(glm::vec3(x, y, int(Settings::Size) + 1));
    }
  }
  for (int x = 0; x < int(Settings::Size); x++) {
    for (int y = 0; y < int(Settings::Size); y++) {
      wall.push_back(glm::vec3(x, y, -1));
    }
  }
}

std::vector<glm::vec3> World::render() { return wall; }
